classes/class-wc-connect-api-client.php

```diff
@@ -475,6 +496,11 @@ protected function request_headers() {
 				return $authorization;
 			}
 
+			$wc_helper_auth_info = WC_Connect_Functions::get_wc_helper_auth_info();
+			if ( is_wp_error( $wc_helper_auth_info ) ) {
```

Comment:
I think your initial take on this was right and for now we'll still send requests even if the Helper auth is missing since we will only need it for certain features limited by a subscription. I'm thinking if the auth isn't found the rest of the plugin should still work, but any feature that requires a subscription shouldn't work. 

With that in mind, I was wondering if this check could just be placed near the end of this function and only include the `X-Woo-Signature` if `$wc_helper_auth_info` is not an error?

Reply:
Yes, this is right - I should have trusted my gut. I added checks for both the signature and the access_token + site_id.